### PR TITLE
added a feature to the rate_estimation function...

### PR DESCRIPTION
**Pull request by rmeyes**
...that calculates the optimized kernel width when the kernel width is specified as 'auto' rather than a fixed value. The method for the optimization of the kernel width was published by H. Shimazaki and S. Shinomoto in 2010 (H. Shimazaki, S. Shinomoto, J Comput Neurosci (2010) 29:171–182.). The Python Implementation is provided as a ready-to-use open source file that can be found on this website: http://toyoizumilab.brain.riken.jp/hideaki/res/kernel.html.
This pull request provides a portable version of the above kernel optimization function compatible with neo and the already existing rate estimation function of elephant.

### CODE REVIEW

File: elephant/statistics.py [hunk 1]
```diff
@@ -629,3 +647,220 @@ def time_histogram(spiketrains, binsize, t_start=None, t_stop=None,
                                  sampling_period=binsize, units=bin_hist.units,
                                  t_start=t_start)
 
+
+"""Kernel Bandwidth Optimization.
+Copyright (c) 2014 Subhasis Ray
```
Comment (apdavison) [inline elephant/statistics.py]: You can't include GPL-licenced code in a BSD-licenced project. Can you ask Subhasis Ray if he is willing to relicence the code? (If you can't find his e-mail address, let me know)

**Comment by rmeyes** [comment]
I'm sorry about this, I don't know much about licensing and what can or cannot be used. I tried to find contact information of Subhasis Ray but wasn't successful. I sent him a message via Google+, as it was the only way possible for me to contact him. Additionally, I sent an email to the author of the original publication, Hideaki Shimazaki, and asked him whether he can provide any contact information. I will let you know as soon as I get a response but in the meantime you may be more successful that I was.

**Comment by rmeyes** [comment]
I am in contact with Subhasis and this is what he replied in his email:

"Dear Richard,
I have no problem with switching to modified BSD license used in  ElePhAnT, which is compatible with gpl2."

Is this email response enough to use the code or are there any other steps on his side that need to be made?

**Comment by apdavison** [comment]
I think the e-mail from Subhasis is good enough. You should now remove the copyright statement and licence information from the docstring, and add Subhasis' name and affiliation (and your own) to `doc/authors.rst`.

**Comment by rmeyes** [comment]
Okay, I made the changes to the author list and removed the license information form the quote. I hope it's good to go now :)

**Comment by rmeyes** [comment]
Oh I see, thanks for the clarification. As I said, I'm not an expert on those licensing issues. Sorry for the confusing commit comment. Thanks for merging!

**Pull request merged**